metal/rtc.h

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2019 SiFive, Inc. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__RTC_H
+#define METAL__RTC_H
+
+#include <stdint.h>
+
+/*!
+ * @file rtc.h
+ * @brief API for Real-Time Clocks
+ */
+
+struct metal_rtc;
+
+/*!
+ * @brief List of RTC run behaviors
+ */
+enum metal_rtc_run_option {
+    METAL_RTC_STOP = 0,
+    METAL_RTC_RUN,
+};
+
+struct metal_rtc_vtable {
+    uint64_t (*get_rate)(const struct metal_rtc *const rtc);
+    uint64_t (*set_rate)(const struct metal_rtc *const rtc, const uint64_t rate);
+    uint64_t (*get_compare)(const struct metal_rtc *const rtc);
+    uint64_t (*set_compare)(const struct metal_rtc *const rtc, const uint64_t compare);
+    uint64_t (*get_count)(const struct metal_rtc *const rtc);
+    uint64_t (*set_count)(const struct metal_rtc *const rtc, const uint64_t count);
+    int (*run)(const struct metal_rtc *const rtc, const enum metal_rtc_run_option option);
+    struct metal_interrupt *(*get_interrupt)(const struct metal_rtc *const rtc);
+    int (*get_interrupt_id)(const struct metal_rtc *const rtc);
+};
+
+/*!
+ * @brief Handle for a Real-Time Clock
+ */
+struct metal_rtc {
+    const struct metal_rtc_vtable *vtable;
+};
+
+/*!
+ * @brief Get the rate of the RTC
+ * @return The rate in Hz
+ */
+inline uint64_t metal_rtc_get_rate(const struct metal_rtc *const rtc) {
```

Comment:
For 32bits system, does not make sense to store or use 54bits value. May be uintptr_ptr type be more appropriate.